diff -u -r zlib-1.2.11.orig/zconf.h zlib-1.2.11/zconf.h
--- zlib-1.2.11.orig/zconf.h	2017-01-01 08:37:10.000000000 +0100
+++ zlib-1.2.11/zconf.h	2020-06-25 20:45:23.893893700 +0200
@@ -329,6 +329,15 @@
 #endif
 
 #if defined(WINDOWS) || defined(WIN32)
+#  if defined(_MSC_VER)
+#    ifdef ZLIB_INTERNAL
+#      include <io.h>
+#      define open _open
+#      define close _close
+#      define read _read
+#      define write _write
+#    endif
+#  endif
    /* If building or using zlib as a DLL, define ZLIB_DLL.
     * This is not mandatory, but it offers a little performance increase.
     */
