diff -u -r bullet3-2.87.orig/src/LinearMath/btVector3.h bullet3-2.87/src/LinearMath/btVector3.h
--- bullet3-2.87.orig/src/LinearMath/btVector3.h	2020-08-25 16:33:13.779545412 +0200
+++ bullet3-2.87/src/LinearMath/btVector3.h	2020-08-25 16:40:45.629254825 +0200
@@ -1105,9 +1105,14 @@
 		mVec128 = vec;
 	}
 
+	SIMD_FORCE_INLINE btVector4(const btVector4& rhs)
+		:	btVector3(rhs.mVec128)
+	{
+	}
+
 	SIMD_FORCE_INLINE btVector4(const btVector3& rhs)
+		:	btVector3(rhs.mVec128)
 	{
-		mVec128 = rhs.mVec128;
 	}
 
 	SIMD_FORCE_INLINE btVector4& 
