diff -ur libiconv-1.16.orig/lib/loop_wchar.h libiconv-1.16/lib/loop_wchar.h
--- libiconv-1.16.orig/lib/loop_wchar.h	2020-08-29 13:20:48.000000000 +0200
+++ libiconv-1.16/lib/loop_wchar.h	2020-08-29 13:58:54.000000000 +0200
@@ -36,7 +36,11 @@
 # include <wchar.h>
 # define BUF_SIZE 64  /* assume MB_LEN_MAX <= 64 */
   /* Some systems, like BeOS, have multibyte encodings but lack mbstate_t.  */
+#ifndef _MSC_VER
   extern size_t mbrtowc ();
+#else
+# define mbsinit(_P)  (_P == NULL || *_P == 0)
+#endif
 # ifdef mbstate_t
 #  define mbrtowc(pwc, s, n, ps) (mbrtowc)(pwc, s, n, 0)
 #  define mbsinit(ps) 1
